examples/cifar10/train_benchmark.py

```diff
@@ -0,0 +1,61 @@
+"""Benchmark for the CIFAR10 example."""
+import tempfile
+
+import time
+from absl import flags
+from absl.testing import absltest
+from absl.testing.flagsaver import flagsaver
+
+import jax
+import numpy as np
+from flax.testing import Benchmark
+
+import train
+
+
+# Parse absl flags test_srcdir and test_tmpdir.
+jax.config.parse_flags_with_absl()
+
+
+FLAGS = flags.FLAGS
+
+
+class CifarTenBenchmark(Benchmark):
+  """Benchmarks for the CIFAR10 Flax example."""
+
+  @flagsaver
+  def test_1x_v100(self):
+    """Run Wide ResNet CIFAR10 on 1x V100 GPUs for 2 epochs."""
+    model_dir = tempfile.mkdtemp()
+    FLAGS.num_epochs = 2
+    FLAGS.arch = 'wrn26_10'
+    FLAGS.model_dir = model_dir
+
+    start_time = time.time()
```

Comment:
We could consider making this work with the following abstraction:

```py
with self.benchmark() as duration, summaries:
  train.main([])
```

(`duration` is what was called `benchmark_time` in your code)

Reply:
I like how this makes the code more concise, however, it makes a few assumptions:
1) That `train.py` always accepts `model_dir` as a flag; and
2) That we always rely on summaries to get the necessary metrics.

* Assumption 2) is currently true, however, I can imagine that for some (really small) examples we may decide to instead rely on parsing their logging output to get the (very simple) metrics.
* Assumption 1) is also currently true for models that output TensorBoard summaries, but again, unclear whether this is how we'll always do things.
* I do see one non-disputable benefit of the construct you proposed - it makes it easy to get the duration (i.e. is a timer scope).


I think we can either:
1) Make a timer-only scope
2) Go ahead with making the two assumptions above, but we should keep in mind that not all examples fall under this abstraction.

Curious to hear your thoughts.

Reply:
OK yes I now agree with you, let's not prematurely make it harder to use the framework more flexibly (like your point about parsing output)

The timer scope could be cute but it's not really that important -- one extra line vs two.